app/views/categories/_form.html.erb

```diff
@@ -34,7 +34,10 @@
       <div class="space-y-2">
         <%= f.select :classification, [["Income", "income"], ["Expense", "expense"]], { label: "Classification" }, required: true %>
         <%= f.text_field :name, placeholder: t(".placeholder"), required: true, autofocus: true, label: "Name", data: { color_avatar_target: "name" } %>
-        <%= f.select :parent_id, categories.pluck(:name, :id), { include_blank: "(unassigned)", label: "Parent category (optional)" } %>
+        <%= f.select :parent_id, categories.pluck(:name, :id), { include_blank: "(unassigned)", label: "Parent category (optional)" }, disabled: category.parent? %>
+        <% if category.parent? %>
+          <span class="text-xs italic pl-2 text-gray-500">This category couldn’t be assigned as a subcategory because it already has subcategories.</span>
+        <% end %>
```

Comment:
I think here, we can just have a conditional:

```erb
<% unless category.parent? %>
  <%= f.select :parent_id ...
<% end %>
```

That way, if the user is not allowed to assign a parent, we don't ever give them the option.

Reply:
Nice, i agree.